### DISCUSSION TITLE
Serverless Function crashing due to `Cannot find module '/var/task/backend/node_modules/axios/dist/node/axios.cjs'`

### DISCUSSION THREAD

**Discussion by cmgchess** [vercel/vercel#9637] [Help] [answered]
Hello 👋
I have a serverless function and it works locally using `vercel dev` but fails when I deploy it.   
[Here](https://github.com/cmgchess/fide-ui/tree/master/backend) is the related repo.
I'm getting the error

```
Cannot find module '/var/task/backend/node_modules/axios/dist/node/axios.cjs'
Did you forget to add it to "dependencies" in `package.json`?
RequestId: f00c415d-275c-4c1e-b6a0-e9546687763e Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

I have the following dependencies including `axios`
```json
  "devDependencies": {
    "@vercel/node": "^2.9.12"
  },
  "dependencies": {
    "axios": "^1.3.3",
    "cheerio": "^1.0.0-rc.12"
  }
```

I use Node 18 on Vercel. I'm not sure what's causing this since I can find the `axios.cjs` file when I check my `node_modules` folder

Help much appreciated 🙏

![image](https://user-images.githubusercontent.com/61736812/224010460-ecb82788-853a-4e51-81a4-39d72679e538.png)
![image](https://user-images.githubusercontent.com/61736812/224009780-dee187fc-dd50-4324-ac2a-fc958a2edaa6.png)
![image](https://user-images.githubusercontent.com/61736812/224010271-8a1de724-6117-433b-ace5-6b11525b6c18.png)

**Comment by cmgchess** [comment, marked as answer]
instead of

```javascript
import axios from 'axios';
```
using

```javascript
import axios from 'axios/dist/node/axios.cjs';
```

solved the issue for me

**EDIT**
This however breaks in dev and only works after deploying

I'm not sure why exactly. Maybe somebody can explain :)

**Comment by cmgchess** [comment]
i also created a reproducible demo
https://github.com/cmgchess/vercel-serverless-axios-issue

**Comment by jetlej** [comment]
I'm getting this error suddenly too! Any updates here?
